github_poster/loader/jike_loader.py

```diff
@@ -134,48 +164,68 @@ def _request_post_data(self, last_id="", resp_date_list=[]):
         r = self.session.post(
             JIKE_GRAPHQL_URL, headers=self.headers, data=json.dumps(payload_data)
         )
+        # print(r.json())
         resp_dates = find_date_in_response(r)
         next_last_id = find_last_id_in_response(r.text)
-        resp_date_list.extend(resp_dates)
+
+        resp_data_list.extend(r.json()["data"]["userProfile"]["feeds"]["nodes"])
         if self._check_if_stop(resp_dates) or next_last_id == "":
-            return resp_date_list
-        self._request_post_data(next_last_id, resp_date_list)
+            return resp_data_list
+        time.sleep(randint(1, 3))
+        self._request_post_data(next_last_id, resp_data_list)
 
-    def _check_if_stop(self, list_of_dates):
+    def _check_if_stop(self, dates):
         """
         stop when content other year data
         """
         r_date = compile(str(self.from_year) + ".*?")
-        filter_list = list(filter(r_date.match, list_of_dates))
-        return len(filter_list) != len(list_of_dates)
+        filter_list = list(filter(r_date.match, dates))
+        if len(filter_list) != len(dates):
+            self.stop_time += 1
+        return self.stop_time == 3
 
     def get_api_data(self):
-        dateTime_cache = []
-        # get first last id
-        first_last_id, first_dateTime_list = self._get_first_last_id()
+        data_cache = []
+        # get first last id and first data
+        first_last_id, first_data_dict = self._get_first_last_id()
         if first_last_id == "":
             raise LoadError("Can not get first last id, please check your cookie")
-        dateTime_cache.extend(first_dateTime_list)
-        if self._check_if_stop(first_dateTime_list):
-            return dateTime_cache
+        if self._check_if_stop(first_data_dict.keys()):
+            return data_cache
         else:
             # do post get more
             post_resp_dates = []
             self._request_post_data(first_last_id, post_resp_dates)
-            dateTime_cache.extend(post_resp_dates)
-            return dateTime_cache
+            data_cache.extend(post_resp_dates)
+            return data_cache, first_data_dict
 
     def make_track_dict(self):
-        data_list = self.get_api_data()
-        for d in data_list:
-            date_str = pendulum.parse(d, tz=self.time_zone).to_date_string()
-            self.number_by_date_dict[date_str] += 1
+        count_types = ("record", "like", "comment", "repost", "share")
+        if self.count_type not in count_types:
```

Comment:
I think we do not check here.
check in `cli` may better?
![image](https://user-images.githubusercontent.com/15976103/147423033-e76efec9-a8a2-46ca-9bbd-7114b507d6de.png)

https://docs.python.org/3/library/argparse.html